provider/cmd/main.go

```diff
@@ -4,18 +4,45 @@
 package main
 
 import (
+	"errors"
 	"fmt"
 	"os"
+	"syscall"
 
 	"github.com/ironcore-dev/libvirt-provider/provider/cmd/app"
+	originzap "go.uber.org/zap"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
+const (
+	errTerminatedApp = "app terminated with error"
+)
+
 func main() {
 	ctx := ctrl.SetupSignalHandler()
 
-	if err := app.Command().ExecuteContext(ctx); err != nil {
-		fmt.Println(err.Error())
-		os.Exit(1)
+	cmd, ptrLogger := app.Command()
+	exitCode := int(0)
```

Comment:
This exclusive conversion of 0 to int is not needed.

Reply:
it isn't retyping in this case. I specified type only. I checked assembly and both format produce same assembly code. If it is confusing for you, I can use var syntax but i prefer define specific type.